contracts/ServiceRegistryL2.sol

```diff
@@ -0,0 +1,878 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import "./GenericRegistry.sol";
+import "./interfaces/IMultisig.sol";
+import "./interfaces/IRegistry.sol";
+
+// This struct is 128 bits in total
+struct AgentParams {
+    // Number of agent instances. This number is limited by the number of agent instances
+    uint32 slots;
+    // Bond per agent instance. This is enough for 1b+ ETH or 1e27
+    uint96 bond;
+}
+
+// This struct is 192 bits in total
+struct AgentInstance {
+    // Address of an agent instance
+    address instance;
+    // Canonical agent Id. This number is limited by the max number of agent Ids (see UnitRegistry contract)
+    uint32 agentId;
+}
+
+/// @title Service Registry L2 - Smart contract for registering services on L2 chains.
+/// @dev Underlying canonical agents and components are not checked for their validity since they are set up on the L1 mainnet.
```

Comment:
```suggestion
/// @dev Underlying canonical agents and components are not checked for their validity since they are set up on the L1 mainnet. The architecture is optimistic, in the sense that service owners are assumed to reference existing and relevant agents.
```